composer/trainer/trainer_hparams.py

```diff
@@ -262,6 +262,8 @@ class TrainerHparams(hp.Hparams):
         grad_clip_norm (float, optional): See :class:`.Trainer`.
 
         profiler (ProfilerHparams, optional): Profiler hyperparameters.
+
+    .. jsonschema:: ../json_schemas/trainer_hparams.json
```

Comment:
Instead of generating a new jsonschema for each hparams class, and linking in the docs to that file, can we link the the definition of that hparams class within the experiment hparams json schema? The experiment hparams is the "root", and all possible hparams should be used within it. The JSONSchema plugin supports JSONPointer notation.

For example:

```suggestion
    .. jsonschema:: ../json_schemas/experiment_hparams.json#/defs/trainer_hparams
```

This should help reduce the duplication, and make the docs site more accessible (since there will be just one jsonschema entry)

Reply:
You're robbing me of my 25k lines committed though :(

Reply:
re offline discussion, we actually do need all the files to enable autocomplete